packages/graphback-runtime-mongodb/src/MongoDBDataProvider.ts

```diff
@@ -83,29 +81,28 @@ export class MongoDBDataProvider<Type = any> implements GraphbackDataProvider<Ty
       throw new NoDataError(`Cannot delete ${this.collectionName} - missing ID field`)
     }
 
-    const objectId = new ObjectId(idField.value);
     const projection = this.buildProjectionOption(context);
+    const objectId = new ObjectId(idField.value);
     const queryResult = await this.db.collection(this.collectionName).findOne({ _id: objectId }, { projection });
     if (queryResult) {
       const result = await this.db.collection(this.collectionName).deleteOne({ _id: objectId });
       if (result.result.ok) {
-
-        return this.mapFields(queryResult);
+        return queryResult;
       }
     }
     throw new NoDataError(`Cannot update ${this.collectionName}`);
   }
 
   public async findOne(filter: any, context: GraphbackContext): Promise<Type> {
-    if (filter.id) {
-      filter = { _id: new ObjectId(filter.id) }
+    if (filter._id) {
+      filter._id = new ObjectId(filter._id);
```

Comment:
If the query is `get<T>(id: GraphbackObjectID)` should this already by an `ObjectID`?

Anyway, it depends on the outcome of whether or not to change the schema `get` query to allow scalar variations, but it is fine as is.

Reply:
Yes. This line is a left over from a previous iteration, it should be gone.